pkg/core/interop/context.go

```diff
@@ -336,12 +336,23 @@ func (c *ContractMD) buildHFSpecificMD(hf config.Hardfork) {
 	if c.onManifestConstruction != nil {
 		c.onManifestConstruction(m)
 	}
+	var updateCounter uint16
+	for _, h := range config.Hardforks {
+		if h > hf {
+			break
+		}
+		if _, ok := c.ActiveHFs[h]; ok {
+			updateCounter++
+		}
+	}
```

Comment:
This won't work if you have two hardforks enabled at the same height for some private installation.

This also won't work if you have all hardforks enabled in the genesis (I'm aware that AxLabs use private installation with all hardforks enabled starting from the genesis).

This issue connected with the way of how we update native contracts. The latest active hardfork is taken to build native state every time it needs to be updated (or deployed):
https://github.com/nspcc-dev/neo-go/blob/31a99d44eb97e24410a7b8600358986f0d3236a5/pkg/core/native/management.go#L620-L625

Reply:
We may use Management to properly handle `getnativecontracts` RPC call.

Reply:
Unfortunately we don't have complete hardfork configuration here.

Reply:
> We may use Management to properly handle getnativecontracts RPC call.

An option. With additional calls. But we have 'em cached, unlike C#. Meh. I wanted to make it better.

Reply:
I know, that's why I'm against of ContractBase removal.